core/types/calls.cc

```diff
@@ -1227,9 +1235,43 @@ TypePtr AppliedType::getCallArguments(const GlobalState &gs, NameRef name) const
     return getMethodArguments(gs, klass, name, targs);
 }
 
+bool MetaType::canCallNew(const GlobalState &gs) const {
```

Comment:
Do you want to add a comment here about how this is conservative, because there are some cases where we can't tell whether to allow it or not?